src/util/file_log.rs

```diff
@@ -57,47 +57,92 @@ fn open_log_file(path: &str) -> io::Result<File> {
     OpenOptions::new().append(true).create(true).open(path)
 }
 
+#[derive(Clone, Copy)]
+enum RotateType {
+    ByDay,
+    BySize,
+}
+
 struct RotatingFileLoggerCore {
-    rollover_time: Tm,
+    rotate_type: RotateType,
+    rotate_size: Option<usize>,
```

Comment:
How about put the `rotate_size` in the `RotateType::BySize(usize)`?

Reply:
And `RotateType::ByDay(Tm)`.